### ISSUE TITLE
src/proxy.c: uninitialized pointer read suspected by coverity

### ISSUE THREAD

**Issue by chipitsine** [haproxy/haproxy#1568]
### Tool Name and Version

coverity

### Code Report

```plain
1837struct proxy *parse_new_proxy(const char *name, unsigned int cap,
1838                              const char *file, int linenum,
1839                              const struct proxy *defproxy)
1840{
1841        struct proxy *curproxy = NULL;
    1. var_decl: Declaring variable errmsg without initializer.
1842        char *errmsg;
1843
      CID 1469674 (#1 of 2): Uninitialized pointer read (UNINIT) [select issue]
    2. Condition !(curproxy = alloc_new_proxy(name, cap, &errmsg)), taking false branch.
1844        if (!(curproxy = alloc_new_proxy(name, cap, &errmsg))) {
1845                ha_alert("parsing [%s:%d] : %s\n", file, linenum, errmsg);
1846                free(errmsg);
1847                return NULL;
1848        }
1849
    3. Condition defproxy, taking true branch.
1850        if (defproxy) {
    CID 1469674 (#2 of 2): Uninitialized pointer read (UNINIT)4. uninit_use_in_call: Using uninitialized value errmsg when calling proxy_defproxy_cpy. [show details]
1851                if (proxy_defproxy_cpy(curproxy, defproxy, &errmsg)) {
1852                        ha_alert("parsing [%s:%d] : %s\n", file, linenum, errmsg);
1853                        free(errmsg);
1854
1855                        ha_free(&curproxy);
1856                        return NULL;
1857                }
1858        }
```

### Output of `haproxy -vv`

```plain
no
```

**Comment by wtarreau** [comment]
Indeed, errmsg ought to be pre-initialised to NULL here!

**Comment by wtarreau** [comment]
Now fixed, should be backported to 2.4. Thanks Ilya!
